frame/elections-phragmen/src/lib.rs

```diff
@@ -536,7 +536,7 @@ impl<T: Trait> Module<T> {
 		if Self::is_voter(who) {
 			Self::votes_of(who)
 				.iter()
-				.all(|v| !Self::is_member(v) && !Self::is_candidate(v).is_ok())
+				.all(|v| !Self::is_member(v) && !Self::is_runner(v) && !Self::is_candidate(v).is_ok())
```

Comment:
Could you add a test that failed without your changes?

Reply:
Expanded a test case `can_detect_defunct_voter` to detect this bug.